db_dtypes/core.py

```diff
@@ -107,38 +102,7 @@ def isna(self):
         return pandas.isna(self._ndarray)
 
     def _validate_scalar(self, value):
-        if pandas.isna(value):
-            return None
-
-        if not isinstance(value, self.dtype.type):
-            raise ValueError(value)
-
-        return value
-
-    def take(
-        self,
-        indices: Sequence[int],
-        *,
-        allow_fill: bool = False,
-        fill_value: Any = None,
-    ):
-        indices = numpy.asarray(indices, dtype=numpy.intp)
-        data = self._ndarray
-        if allow_fill:
-            fill_value = self._validate_scalar(fill_value)
-            fill_value = (
-                numpy.datetime64() if fill_value is None else self._datetime(fill_value)
-            )
-            if (indices < -1).any():
-                raise ValueError(
-                    "take called with negative indexes other than -1,"
-                    " when a fill value is provided."
-                )
-        out = data.take(indices)
-        if allow_fill:
-            out[indices == -1] = fill_value
-
-        return self.__class__(out)
+        return self._datetime(value)
```

Comment:
From what I'm seeing in https://github.com/pandas-dev/pandas/blob/ad190575aa75962d2d0eade2de81a5fe5a2e285b/pandas/core/arrays/_mixins.py#L156

"_validate_scalar" is probably the wrong name. I appears to be expected that you convert the scalar into something compatible. "_convert_scalar" is probably more appropriate.

Reply:
See also: https://github.com/pandas-dev/pandas/blob/4f712c1c48803650e2be8a3bac9d3cfc047b3da6/pandas/core/arrays/datetimelike.py#L597

`DatetimeLikeArrayMixin` might even be a more appropriate base, but not sure if it's necessary.